src/test/java/uk/gov/justice/hmpps/casenotes/utils/JwtAuthHelper.kt

```diff
@@ -0,0 +1,46 @@
+package uk.gov.justice.hmpps.casenotes.utils
+
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import org.springframework.context.annotation.Bean
+import org.springframework.security.oauth2.jwt.JwtDecoder
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder
+import org.springframework.stereotype.Component
+import java.security.KeyPair
+import java.security.KeyPairGenerator
+import java.security.interfaces.RSAPublicKey
+import java.time.Duration
+import java.util.*
+
+
+@Component
+class JwtAuthHelper {
+  private val keyPair: KeyPair
+
+  init {
+    val gen = KeyPairGenerator.getInstance("RSA")
+    gen.initialize(2048)
+    keyPair = gen.generateKeyPair()
+  }
+
+  @Bean
+  fun jwtDecoder(): JwtDecoder = NimbusJwtDecoder.withPublicKey(keyPair.public as RSAPublicKey).build()
```

Comment:
generate our own decoder for tests and use that instead of hardcoded properties in yaml file